tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc

```diff
@@ -1706,15 +1718,31 @@ void BuildAndRunConvertedNetwork(const string& name, OpConverterTest* test,
   const DataVec input_data{
       {"input", test->AsTensor<T>(CastTestVector<float, T>(input_vec), shape)}};
   DataVec output_data{{name, test->ConstructTensor<T>(6)}};
-  // we have to set the datatype arg correctly
   test->BuildAndRun(input_data, &output_data);
   // Check the shape of the actual output tensor
   TF_EXPECT_OK(TensorShapeUtils::MakeShape(p.expected_output_dims, &shape));
   EXPECT_TRUE(output_data[0].tensor.shape() == shape)
       << "Expected shape: " << shape.DebugString() << ", actual shape"
       << output_data[0].tensor.shape().DebugString();
-  EXPECT_THAT(GetSpanForData<T>(output_data[0]),
-              ElementsAreArray(CastTestVector<float, T>(output_vec)));
+  // Cast the output to float and compare to expected output
+  auto out_span = GetSpanForData<T>(output_data[0]);
+  std::vector<float> casted_output(out_span.begin(), out_span.end());
+  EXPECT_THAT(casted_output, matcher);
+}
+
+void InstantiateBuildAndRun(DataType tf_dtype, const string& name,
+                            OpConverterTest* test, const TestParamBase& p,
+                            const std::vector<float>& input_vec,
+                            const Matcher<std::vector<float>>& matcher) {
+  if (tf_dtype == DT_FLOAT) {
+    BuildAndRunConvertedNetwork<DT_FLOAT>(name, test, p, input_vec, matcher);
+  } else if (tf_dtype == DT_HALF) {
```

Comment:
The branches for HALF AND INT32 are not dead code because we only test DT_FLOAT. Do you actually mean DT_INT8 not DT_INT32?